lib/router/routes.dart

```diff
@@ -15,131 +12,90 @@ import '../login_state.dart';
 import '../ui/login.dart';
 import '../ui/details.dart';
 
-const String loginRoute = '/login';
-const String createAccountRoute = '/create_account';
-const String homeRoute = '/';
-const String mainPageRoute = '/main/:index';
-const String personalPageRoute = 'personal';
-const String paymentPageRoute = 'payment';
-const String signinInfoPageRoute = 'signin';
-const String moreInfoPageRoute = 'moreInfo';
-const String profilePageRoute = 'profile';
-const String cartPageRoute = 'cart';
-const String shoppingPageRoute = 'shopping';
-const String detailsPageRouteName = 'details';
-const String detailsPageRoute = '/details/:item';
-
 class MyRouter {
   final LoginState loginState;
 
   MyRouter(this.loginState);
 
   late final router = GoRouter(
-    initialLocation: homeRoute,
     refreshListenable: loginState,
     debugLogDiagnostics: true,
     urlPathStrategy: UrlPathStrategy.path,
     routes: [
       GoRoute(
-        name: loginPage,
-        path: loginRoute,
+        name: homeRouteName,
+        path: '/',
+        redirect: (state) =>
+            state.namedLocation(mainRouteName, params: {'tab': 'shop'}),
+      ),
+      GoRoute(
+        name: loginRouteName,
+        path: '/login',
         pageBuilder: (context, state) => MaterialPage<void>(
           key: state.pageKey,
           child: const Login(),
         ),
       ),
       GoRoute(
-        name: createAccountPage,
-        path: createAccountRoute,
+        name: createAccountRouteName,
+        path: '/create-account',
         pageBuilder: (context, state) => MaterialPage<void>(
           key: state.pageKey,
           child: const CreateAccount(),
         ),
       ),
       GoRoute(
-        path: homeRoute,
-        redirect: (_) => '/main/0',
-      ),
-      GoRoute(
-          name: mainPage,
-          path: mainPageRoute,
-          pageBuilder: (context, state) {
-            final index = state.params['index']!;
-            return MaterialPage<void>(
+        name: mainRouteName,
+        path: '/main/:tab(shop|cart|profile)',
```

Comment:
Is this a way to show the possible values?

Reply:
yes plus it constrains the values -- everything inside the parens is a regular expression.